### ISSUE TITLE
Which dataset (training/testing) should be used to build the explainer?

### ISSUE THREAD

**Issue by xiaochi-liu** [pbiecek/ema#54]
Hello @pbiecek,

Many thanks for your great work. Recently I'm learning how to use `DALEX` and `DALEXtra` packages to conduct XAI analysis. May I ask you a question about the `explain()` function?

When doing machine learning, we first split the whole dataset into the training and testing data. I have seen most of the tutorials are using the testing data to build the explainer (for example [this one](https://raw.githack.com/pbiecek/DALEX_docs/master/vignettes/DALEX_h2o.html)). This is appropriate when we want to evaluate the model's overall performance.

However, if the analysis purpose is to investigate the relationship between the input features and the target variable (i.e., feature importance, feature effects, and feature interaction), should we use **the training data** to build the explainer?

Your kind guidance is much appreciated!

Best regards,
Xiaochi

**Comment by pbiecek** [comment]
Great question,

Moreover, the choice of distribution for calculating explanations is very important, see: https://arxiv.org/abs/2105.13787

Unfortunately there is no one-size-fits-all answer either, here are some suggestions:
- if you are exploring the model trying to understand the relationships in the data, then as for EDA it is best to use the entire dataset (i train and test). 
- if you want to assess the quality of the model, or have conclusions related to model generalization, it is best to use test data for explanations
- if you want to assess whether model-drift is occurring, it is best to have an out-of-time sample and count explanations on it, comparing them with explanations on training data (in DALEX you can compare explanations for several data sets, see contrastive explanations https://ema.drwhy.ai/partialDependenceProfiles.html#contrastive-partial-dependence-profiles
- it is always a good idea to see if the explanations on the test and training sets are similar, in case the distribution of the test set differs significantly (again using contrastive explanations https://ema.drwhy.ai/partialDependenceProfiles.html#contrastive-partial-dependence-profiles)

To summarize:
- for exploration use the whole data you have
- for model verification count the explanations on different subsets and compare them on one plot using contrastive explanations

**Comment by xiaochi-liu** [comment]
Get it. That's very clear. 

Thank you very much for your kind guidance @pbiecek. I really appreciate it!
